docs/driver-only-builds.md

```diff
@@ -285,10 +285,10 @@ algorithm/mode you can:
   - `MBEDTLS_CHACHAPOLY_C`
   - `MBEDTLS_NULL_CIPHER`
 
-Once a key type and related algorithm are accelerated, all cipher and AEADs
-operations of that type requested through the PSA Crypto API are performed by
-the driver. Only functions belonging to legacy modules which are disabled won't
-be available in this configuration.
+Once a key type and related algorithm are accelerated, all the PSA Crypto APIs
+will work, as well as X.509 and TLS (with MBEDTLS_USE_PSA_CRYPTO enabled) but
+some non-PSA APIs will be absent or have reduced functionality, see
+[Disabling CIPHER_C](#disabling-cipher_c) for details.
```

Comment:
I don't think that's about disabling `CIPHER_C`. Even if we don't disable `CIPHER_C` we don't be getting PEM decryption for example, and I don't think we'll be getting en/decryption support from PKCS(5|12) either.

At this point I'm very tempted to request a new component with `CIPHER_C` not disabled in the accelerated component (we can re-use the same reference component) just to make sure we really understand what the differences are between disabling `CIPHER_C` and leaving it enabled.

Reply:
Actually if we accelerate (and disable the built-ins) for all ciphers, then even if we leave `CIPHER_C` enabled it will be pretty useless. So, the only case where leaving `CIPHER_C` enabled really makes sense is if you have some ciphers driver-only and some of them built-in. Perhaps we should have a component testing that? (But then that's probably out of scope for this PR.)

I'll note a pattern here: for ECC we also forgot to test with some curves accelerated and others not, and got some surprises when we tried it out. Here we also forgot about testing some ciphers accelerated an others not until the end. Sigh.

Reply:
> I'll note a pattern here: for ECC we also forgot to test with some curves accelerated and others not, and got some surprises when we tried it out. Here we also forgot about testing some ciphers accelerated an others not until the end. Sigh.

Do you think it's worth to add a new task to the ciphers/AEADs EPIC?

Reply:
Yes, I think we should. I need to think a bit about exactly what it is that we want to test.

Reply:
Please still fix the link: see "Restrictions" below for details. We need to clearly distinguish:
- What's related to driver-only (regardless of whether `CIPHER_C` is enabled) which goes in the "Restrictions" section.
- What's related to removing `CIPHER_C` (which can happen even without drivers), which goes in the section about that.